src/graph/TraverseExecutor.cpp

```diff
@@ -139,5 +146,99 @@ VariantType Collector::getProp(const std::string &prop,
     }
 }
 
+void Collector::getSchema(const std::vector<VariantType> &vals,
+                          const std::vector<std::string> &colNames,
+                          const std::vector<nebula::cpp2::SupportedType> &colTypes,
+                          SchemaWriter *outputSchema) {
+    DCHECK(outputSchema != nullptr);
+    DCHECK_EQ(vals.size(), colNames.size());
+    DCHECK_EQ(vals.size(), colTypes.size());
+    using nebula::cpp2::SupportedType;
+    auto index = 0u;
+    for (auto &it : colTypes) {
+        SupportedType type;
+        if (it == SupportedType::UNKNOWN) {
+            switch (vals[index].which()) {
+                case VAR_INT64:
+                    // all integers in InterimResult are regarded as type of INT
+                    type = SupportedType::INT;
+                    break;
+                case VAR_DOUBLE:
+                    type = SupportedType::DOUBLE;
+                    break;
+                case VAR_BOOL:
+                    type = SupportedType::BOOL;
+                    break;
+                case VAR_STR:
+                    type = SupportedType::STRING;
+                    break;
+                default:
+                    LOG(FATAL) << "Unknown VariantType: " << vals[index].which();
```

Comment:
Why do I need FATAL here? I don't think FATAL is needed as long as it can be handled internally.